plugins/modules/packaging/os/snap.py

```diff
@@ -227,59 +214,32 @@ def state_present(self):
                   "error output for more details.".format(cmd=self.vars.cmd)
         raise ModuleHelperException(msg=msg)
 
-    def state_absent(self):
-        self.validate_input_snaps()  # if snap doesnt exist, it will be absent by definition
-        actionable_snaps = [s for s in self.vars.name if self.is_snap_installed(s)]
+    def _generic_state_action(self, actionable_func, actionable_var, params=None):
+        actionable_snaps = [s for s in self.vars.name if actionable_func(s)]
         if not actionable_snaps:
             return
         self.changed = True
-        self.vars.snaps_removed = actionable_snaps
+        self.vars[actionable_var] = actionable_snaps
         if self.module.check_mode:
             return
-        params = ['classic', 'channel', 'state']  # get base cmd parts
+        if params is None:
+            params = ['state']
         commands = [params + [{'actionable_snaps': actionable_snaps}]]
         self.vars.cmd, rc, out, err = self._run_multiple_commands(commands)
         if rc == 0:
             return
-        msg = "Ooops! Snap removal failed while executing '{cmd}', please examine logs and " \
+        msg = "Ooops! Snap operation failed while executing '{cmd}', please examine logs and " \
               "error output for more details.".format(cmd=self.vars.cmd)
         raise ModuleHelperException(msg=msg)
 
+    def state_absent(self):
+        self._generic_state_action(self.is_snap_installed, "snaps_removed")
```

Comment:
Shouldn't this be:
```suggestion
        self._generic_state_action(self.is_snap_installed, "snaps_removed", ['classic', 'channel', 'state'])
```
?

Reply:
Nope. The --classic and --channel are not applicable for those commands.

Reply:
The old code did add them, though: https://github.com/ansible-collections/community.general/blob/7b9687f75885f168de52f935dc36f1d35b8b7ed7/plugins/modules/packaging/os/snap.py#L239

Reply:
Well, I can try to do more git archeology on the snap codebase to make sure, but it seems to me that these options were never there for those commands.

Reply:
If you really want to remove them, I would do that in a follow-up PR with its own changelog fragment, but not here.

Reply:
IMHO that's more than needed, but will be happy to go that way.